File: concrete/src/main/java/pcb/uwu/unit/quantity/MinuteUnit.java
package pcb.uwu.unit.quantity;

import pcb.uwu.units.quantity.TimeUnit;

import java.math.BigDecimal;

import static java.math.MathContext.DECIMAL64;

public class MinuteUnit extends TimeUnit {

	private static final BigDecimal SECONDS_PER_MINUTE = new BigDecimal(60);

	public static final MinuteUnit MINUTE = new MinuteUnit();

	private MinuteUnit() {
		super("min", "minute", "minutes",
				value -> value.multipliedBy(SECONDS_PER_MINUTE, DECIMAL64),
				value -> value.dividedBy(SECONDS_PER_MINUTE, DECIMAL64));
	}
}

File: implementation/src/main/java/pcb/uwu/unit/base/MinuteUnit.kt
package pcb.uwu.unit.base

import pcb.uwu.unit.base.MinuteUnit.CONVERT

object MinuteUnit
    : TimeUnit(symbol = "min",
               singularName = "minute",
               pluralName = "minutes",
               translationToCanonical = { it * CONVERT },
               translationFromCanonical = { it / CONVERT })
{
    const val CONVERT = 60
}
